### PR TITLE
Add translation status page & `i18n` labeler

### PR DESCRIPTION
**Pull request by Waxer59**
#### What kind of changes does this PR include?

- Changes to Starlight code
- Something else!

#### Description

- What does this PR change? Give us a brief description.

Added `i18n` tagger and translation status page

Screenshot from 2023-06-10 11-31-57

> An important thing to know when testing is that the PR's section does not find any PR's because there is no PR with the tag `i18n`.

### CODE REVIEW

File: scripts/lib/translation-status/template.html [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+	<meta charset="utf-8" />
+	<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
+	<meta name="theme-color" content="hsl(256, 44%, 15%)" media="(prefers-color-scheme: dark)" />
+	<meta name="theme-color" content="hsl(273, 37%, 93%)" media="(prefers-color-scheme: light)" />
+
+	<title>Starlight Docs Translation Status</title>
+	<meta name="description"
+		content="Translation progress tracker for the Astro Docs site. See how much has been translated in your language and get involved!" />
+	<link rel="canonical" href="https://i18n.docs.astro.build/" />
+	<meta property="og:title" content="Astro Docs Translation Status" />
```
Comment (dreyfus92) [inline scripts/lib/translation-status/template.html:14]:
```suggestion
	<meta property="og:title" content="Starlight Docs Translation Status" />
```

File: scripts/lib/translation-status/template.html [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+<head>
+	<meta charset="utf-8" />
+	<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
+	<meta name="theme-color" content="hsl(256, 44%, 15%)" media="(prefers-color-scheme: dark)" />
+	<meta name="theme-color" content="hsl(273, 37%, 93%)" media="(prefers-color-scheme: light)" />
+
+	<title>Starlight Docs Translation Status</title>
+	<meta name="description"
+		content="Translation progress tracker for the Astro Docs site. See how much has been translated in your language and get involved!" />
+	<link rel="canonical" href="https://i18n.docs.astro.build/" />
```
Comment (dreyfus92) [inline scripts/lib/translation-status/template.html:13]: Not sure which will be the subdomain this might change.
```suggestion
	<link rel="canonical" href="https://i18n.starlight.docs.astro.build/" />
```

File: scripts/lib/translation-status/template.html [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+<head>
+	<meta charset="utf-8" />
+	<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
+	<meta name="theme-color" content="hsl(256, 44%, 15%)" media="(prefers-color-scheme: dark)" />
+	<meta name="theme-color" content="hsl(273, 37%, 93%)" media="(prefers-color-scheme: light)" />
+
+	<title>Starlight Docs Translation Status</title>
+	<meta name="description"
+		content="Translation progress tracker for the Astro Docs site. See how much has been translated in your language and get involved!" />
```
Comment (dreyfus92) [inline scripts/lib/translation-status/template.html:12]:
```suggestion
		content="Translation progress tracker for the Starlight Docs site. See how much has been translated in your language and get involved!" />
```

File: scripts/lib/translation-status/template.html [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+	<link rel="canonical" href="https://i18n.docs.astro.build/" />
+	<meta property="og:title" content="Astro Docs Translation Status" />
+	<meta property="og:type" content="website" />
+	<meta property="og:url" content="https://i18n.docs.astro.build/" />
```
Comment (dreyfus92) [inline scripts/lib/translation-status/template.html:16]: Here happens the same thing as previously mentionned.
```suggestion
	<meta property="og:url" content="https://i18n.starlight.docs.astro.build/" />
```

File: scripts/lib/translation-status/template.html [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+	<meta property="og:description"
+		content="Translation progress tracker for the Astro Docs site. See how much has been translated in your language and get involved!" />
```
Comment (dreyfus92) [inline scripts/lib/translation-status/template.html:18]:
```suggestion
		content="Translation progress tracker for the Starlight Docs site. See how much has been translated in your language and get involved!" />
```

**Review by dreyfus92 — CHANGES_REQUESTED** [review]
We just need to ask @delucis what will the subdomain for the tracker will be, aside from that I leave you a couple of suggestions, excellent job @Waxer59

**Comment by Waxer59** [comment]
Thanks for the review 🤠

**Comment by delucis** [comment]
I think we could use `https://i18n.starlight.astro.build/` similar to the pattern we use for the docs site. What do you think?

**Comment by Waxer59** [comment]
I think `https://i18n.starlight.astro.build/` is perfect

**Comment by Waxer59** [comment]
I have added Italian to the page

**Review by ElianCodes — APPROVED** [review]
If @delucis is good with the Prettier addition, LGTM!

File: src/i18n/languages.ts
Comment (delucis) [inline src/i18n/languages.ts:1]: I’d suggest either moving this into `scripts/translations-status.ts` directly given we only use it there. Or — even better — figure out how to share the `locales` data from `astro.config.mjs` instead of duplicating it.

File: .prettierrc
Comment (delucis) [inline .prettierrc:1]: I think we can remove this from the PR. Happy do discuss if we need Prettier in this repo, but would prefer to do it in a separate PR so we can stay focused on one thing at a time.

**Review by delucis — COMMENTED** [review]
Thanks again for this @Waxer59! Sorry it took me a few days to get to this.

Left a few comments. Let me know if anything isn’t clear or if there’s anything you prefer I do!

File: .prettierrc
Comment (Waxer59) [reply .prettierrc:1]: Ok, no problem, I added it only so that the text would have a format to read it better when modifying the code.

File: scripts/lib/translation-status/template.html [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+				display: none;
+			}
+		}
+	</style>
+</head>
+
+<body>
+	<main class="main-container">
+		<img src="https://starlight.astro.build/_astro/hero-star.81c1bfde_kyCK2.webp" alt="Starlight logo"
+			class="bg-img">
```
[the remark this answers is not in the document]
Comment (Waxer59) [reply scripts/lib/translation-status/template.html:267]: image
I agree, I have modified the page by adding a new bg and changing the color of the links, what do you think?

File: src/i18n/languages.ts
Comment (Waxer59) [reply src/i18n/languages.ts:1]: Ok, what I would do for the moment to not delay too much this PR would be to move this file inside `translation-status.ts` for the moment until we figure out a way to share this data.

**Comment by Waxer59** [comment]
I have already added the suggestions from the review 🚀

File: scripts/lib/translation-status/template.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (delucis) [reply scripts/lib/translation-status/template.html]: Nice! Yeah, looks good. Thank you.

**Comment by delucis** [comment]
Thanks @Waxer59! Closed my computer for the day but will take a look tomorrow morning. Great work 💖

**Comment by Waxer59** [comment]
In the end I have removed the duplicity of the languages for `translation-status` and used the local ones from `astro.config.mjs`. Check it tomorrow when you can!

File: docs-i18n-tracker/lib/translation-status/template.html [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+	<main class="main-container">
+		<div class="limit-to-viewport">
+			<h1>Starlight Docs Translation Status</h1>
+
+			<p>
+				If you're interested in helping us translate
+				<a href="https://starlight.astro.build/">starlight.astro.build</a> into one of the languages listed
+				below, you've come to the right place! This auto-updating page always lists all the
+				content that could use your help right now.
+			</p>
```
Comment (sarah11918) [inline docs-i18n-tracker/lib/translation-status/template.html:246]:
```suggestion
			<p>
				If you're interested in helping us translate
				<a href="https://starlight.astro.build/">starlight.astro.build</a> into one of the languages listed
				below, you've come to the right place! This auto-updating page always lists all the
				content that could use your help right now.
			</p>
			<p>
				Before starting a new translation, be sure to check the [existing Starlight PRs](https://github.com/withastro/starlight/pulls) to see if this page has already been translated, and consider reviewing any open PRs in your language!
			</p>
```
Thoughts? Maybe we should add something like this to Astro docs translation status page, too!

**Review by sarah11918 — APPROVED** [review]
The preview looks great to me! Just left an idea we could add!

**Comment by Waxer59** [comment]
image

**Review by delucis — APPROVED** [review]
I think this is ready to go! Thanks to everyone who reviewed and thanks to @Waxer59 for putting this together! 🚀
